src/screens/FilterScreen.js

```diff
@@ -1,34 +1,198 @@
+        const nextSetOfSites = nextProps.displaySites;
+        const currentSites = this.props.displaySites;
+
+
+        if (nextSetOfSites.length !== currentSites.length) {
+            this.props.navigation.setParams({siteCount: nextSetOfSites.length});
+        }
+    }
+
+    static renderRightNavButton = (navigate, {siteCount}) => {
+        if (Platform.OS === 'ios') {
+            return (
+                <Button
+                    title={_.isUndefined(siteCount) ? '' : `${siteCount} Results`}
+                    onPress={() => navigate(navKeys.SEARCH)}
+                    backgroundColor="rgba(0,0,0,0)"
+                    color="rgba(0,122,255,1)"
+                />
+            );
+        } else if (Platform.OS === 'android') {
+            // android-specific code for navigation here
+        }
+    };
+
     static navigationOptions = (props) => {
-        const {navigation: {navigate}} = props;
+        const {navigation: {navigate, state: {params = {}}}} = props;
 
         return {
-            headerTitle: 'Filter Your Search'
+            headerRight: FilterScreen.renderRightNavButton(navigate, params)
         }
 
     };
 
+    renderHeader = (section, index, isActive) => {
+        const {headerStyle, headerTextStyle} = styles;
 
-    render(){
-        return(
-            <Card>
-                <CardSection>
-                    <Text>
-                        Filter your sites
-                    </Text>
-                </CardSection>
-            </Card>
+        return (
+            <View style={headerStyle}>
+                <Text style={headerTextStyle}>
+                    {section.title}
+
+                </Text>
+                <Icon type='entypo' name={isActive ? 'chevron-down' : 'chevron-up'} size={25} color={campsiteIcon}/>
+            </View>
+        );
+    };
+
+    onClickCheckbox = (key) => {
+        this.props.checkboxWasClicked({filterKey: key})
+    };
+
+    renderCheckedState = (key) => {
+        const {filterCriteriaKeys} = this.props;
+        const flatFilterCriteriaList = _(filterCriteriaKeys)
+            .map(categoryKeys => {
+                return _.map(categoryKeys, innerKey => innerKey)
+            })
+            .flatten()
+            .valueOf();
+
+        return _.includes(flatFilterCriteriaList, key);
+    };
+
+    renderCheckboxes = (checkboxObject) => {
+        return _.map(checkboxObject, (value, key) => {
+            return (
+                <CheckBox
+                    key={key}
+                    title={value}
+                    checked={this.renderCheckedState(key)}
+                    onPress={() => this.onClickCheckbox(key)}
+                />
+            );
+        })
+    };
+
+    renderContent = (section) => {
+        const {contentStyle, contentTextStyle} = styles;
+
+        return (
+            <View style={contentStyle}>
+                {this.renderCheckboxes(section.content)}
+            </View>
```

Comment:
This `renderContent` function is another method that comes with `react-native-collapsible`. @amgoodman5 feel free to change any styling you want anywhere in this file to make things look better.